### ISSUE TITLE
False positive: not covered mutant for arrow function

### ISSUE THREAD

**Issue by Bilge** [infection/infection#1747]
| Question    | Answer
| ------------| ---------------
| Infection version | 0.26.16
| Test Framework version | PHPUnit 9.5.26
| PHP version | PHP 8.1.3
| Platform    | Windows
| Github Repo | https://github.com/ScriptFUSION/Porter

The following code will generate a _not covered mutant_.

```php
        $this->clearTransformers()->addTransformers(array_map(
            static fn ($transformer) => clone $transformer,
            $transformers
        ));
```

<details>

```
Not Covered mutants:
====================

1) C:\Users\Bilge\Documents\PhpstormProjects\Porter\src\Specification\Specification.php:48    [M] CloneRemoval

--- Original
+++ New
@@ @@
     {
         $this->resource = clone $this->resource;
         $transformers = $this->transformers;
-        $this->clearTransformers()->addTransformers(array_map(static fn($transformer) => clone $transformer, $transformers));
+        $this->clearTransformers()->addTransformers(array_map(static fn($transformer) => $transformer, $transformers));
         \is_object($this->context) && ($this->context = clone $this->context);
         isset($this->recoverableExceptionHandler) && ($this->recoverableExceptionHandler = clone $this->recoverableExceptionHandler);
```

</details>

However, this code will **not** generate a _not covered mutant_. Note that it is the same code, just expressed with a long-form lambda function instead of a single-line arrow function.

```php
        $this->clearTransformers()->addTransformers(array_map(
            static function ($transformer) {
                return clone $transformer;
            },
            $transformers
        ));
```

Suffice it to say, removing the `clone` keyword as claimed by the Infection output to be an _not covered mutant_, causes the tests to fail in both cases.

**Comment by sanmai** [comment]
Suspected duplicate of https://github.com/infection/infection/issues/1736

Infection uses coverage reports provided by a testing framework such as PHPUnit. If it says a certain line isn't covered, that's what it is.

**Comment by sanmai** [comment]
Related to #1750.

**Comment by maks-rafalko** [comment]
This should have been fixed in #1750. If not please reopen
